samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -569,6 +570,138 @@ def _enrich_image_lambda_function(
         cfn_lambda_function["Metadata"]["DockerBuildArgs"] = cfn_docker_build_args
 
 
+def _enrich_lambda_layer(
+    sam_metadata_resource: Dict,
+    cfn_lambda_layer: Dict,
+    cfn_lambda_layer_logical_id: str,
+    terraform_application_dir: str,
+    output_directory_path: str,
+) -> None:
+    """
+    Use the sam metadata resources to enrich the lambda layer.
+
+    Parameters
+    ----------
+    sam_metadata_resource: Dict
+       The sam metadata resource properties
+    cfn_lambda_layer: dict
+       CloudFormation lambda layer to be enriched
+    cfn_lambda_layer_logical_id: str
+       the cloudFormation lambda layer to be enriched logical id.
+    output_directory_path: str
+       the output directory path to write the generated metadata and makefile
+    terraform_application_dir: str
+       the terraform project root directory
+    """
+    sam_metadata_attributes = sam_metadata_resource.get("values", {}).get("triggers", {})
+    sam_metadata_resource_address = sam_metadata_resource.get("address")
+    if not sam_metadata_resource_address:
+        raise PrepareHookException(
+            "Invalid Terraform plan output. The address property should not be null to any terraform resource."
+        )
+    _validate_referenced_resource_layer_matches_metadata_type(
+        cfn_lambda_layer, sam_metadata_attributes, sam_metadata_resource_address
+    )
+    LOG.info(
+        "Enrich the Lambda Layer Version %s using the metadata properties defined in resource %s",
+        cfn_lambda_layer_logical_id,
+        sam_metadata_resource_address,
+    )
+
+    cfn_source_code_path = _get_source_code_path(
+        sam_metadata_attributes,
+        sam_metadata_resource_address,
+        terraform_application_dir,
+        "original_source_code",
+        "source_code_property",
+        "source code",
+    )
+
+    _set_zip_metadata_resources(
+        cfn_lambda_layer,
+        cfn_source_code_path,
+        output_directory_path,
+        terraform_application_dir,
+        CFN_CODE_PROPERTIES[CFN_AWS_LAMBDA_LAYER_VERSION],
+    )
+
+
+def _validate_referenced_resource_layer_matches_metadata_type(
```

Comment:
I see that you unit test this function as part of the `_enrich_lambda_layer` .. I think it is better to have separate unit test cases for it, and also to unit test the exception scenarios